virtcontainers/vm.go

```diff
@@ -99,12 +94,6 @@ func NewVM(ctx context.Context, config VMConfig) (*VM, error) {
 		if err != nil {
 			return nil, err
 		}
-
-		virtLog.WithField("vm id", id).Info("check agent status")
-		err = agent.check()
```

Comment:
Sorry, but you cannot simply remove this. The template factory needs the check to make sure agent is fully up and running before taking a snapshot of the vm. It ensures that we share the initial states of initramfs, guest kernel and kata-agent among all the guests created by the template factory.

Let's take a step back and look at the whole picture. -- You need to hotplug all comm channels, but vm factory requires a working agent. So could you add comm channel hotplug to `NewVM()` instead? Then it still returns a fully working guest and you do not need to split factory's `GetVM` API.